### PR TITLE
Add Tuya TS0202 (_TZ3000_bsvqrxru) PIR motion sensor quirk

### PR DESCRIPTION
**Pull request by vjonas**
## Summary

Add a ZHA quirk for the Tuya **HW500A** PIR motion sensor (`_TZ3000_bsvqrxru` / `TS0202`). Out of the box ZHA renders this device as a `binary_sensor.<name>_opening` (contact switch) that never updates. The quirk fixes it so HA gets a proper `binary_sensor.<name>` motion entity plus a configurable `number.<name>_motion_clear_timeout`.

### Why

The HW500A is a cheap, widely-available Tuya PIR sensor. It ships with three stock-ZHA annoyances:

1. It never reports `zone_type`, so ZHA defaults the binary_sensor to `device_class: opening`.
2. Its firmware stays radio-silent until the "Tuya spell" (a read of specific Basic-cluster attributes) is cast — the PIR blinks its red LED locally on motion, but nothing leaves the device until the spell runs.
3. It exposes a vestigial `OnOff` client out-cluster that ZHA turns into a second, permanently-off binary_sensor.

### What the quirk does

* Pins `zone_type = Motion_Sensor` so the binary_sensor renders as `device_class: motion` without waiting for a `read_attributes` roundtrip.
* Uses `TuyaQuirkBuilder.tuya_enchantment()` to cast the Basic-cluster spell at device init, matching z2m's behaviour for this fingerprint.
* Adds a client-side auto-clear timer (default 90 s, min 5 s, max 900 s) — the HW500A's on-device `keep_time` is firmware-hardcoded (~65 s) and not configurable over Zigbee, which is why z2m's `_TZ3000_bsvqrxru` definition has no `keep_time` expose. Surfaces that timer as a writable `motion_timeout` synthetic attribute backed by a Number entity (HA `entity_type: CONFIG`, step 5 s, unit `s`). Writes stay quirk-local — no Zigbee frames are emitted, we just update the reset timer the next time a motion event fires.
* Handles **both** paths firmwares of the TS0202 family use to report motion: `zone_status_change_notification` commands (inherited from `MotionWithReset`) **and** `zone_status` attribute reports (z2m uses `fz.ias_occupancy_alarm_1` for the former and `fz.ias_occupancy_alarm_1_report` for the latter).
* Replaces stock `PowerConfiguration` with `TuyaPowerConfigurationCluster2AAA` (these units run on 2x AAA).
* Removes the vestigial `OnOff` client out-cluster.

### Zigbee2MQTT reference

* [`TuYa HW500A` definition](https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/src/devices/tuya.ts) — added in [Koenkk/zigbee-herdsman-converters#6230](https://github.com/Koenkk/zigbee-herdsman-converters/pull/6230), `fz.ias_occupancy_alarm_1` + `fz.ias_occupancy_alarm_1_report`, no `keep_time` expose (firmware-hardcoded), no tamper expose (no physical tamper switch on the HW500A).

## Tests

* **22 new unit tests** in `tests/test_tuya_ts0202.py` covering:
  - Quirk application (`zone_type` pinned to `Motion_Sensor`, synthetic `motion_timeout` attribute declared & primed, `OnOff` client cluster removed, `PowerConfiguration` swapped for the 2xAAA variant).
  - Motion detection via both paths: `zone_status` attribute reports (Alarm_1 + Alarm_2) and `zone_status_change_notification` commands.
  - Auto-clear timer scheduling, cancellation, and release.
  - Writable `motion_timeout`: by name, by id, clamping above max, clamping below min, propagating into the next auto-clear timer.
  - Mixed writes that combine synthetic + real attrs (synthetic stays local, real ones are forwarded to the device with `manufacturer=None`).
  - Key-classifier edge cases (`ZCLAttributeDef` input, non-numeric tuples, etc.).
* **Patch coverage: 100%** on `zhaquirks/tuya/ts0202.py`.
* Full `pytest tests/` pass: **4771 passed, 2 xfailed** (no regressions).

## Tested with

Two physical HW500A devices running firmware `0x00000046`, joined via a Texas Instruments CC2652 coordinator on ZHA. Verified:

* Device enrols as IAS Zone and binary_sensor reports motion correctly.
* Setting `number.<name>_motion_clear_timeout` via the HA UI clamps + persists the value and the next motion event clears after the new interval.
* No spurious `binary_sensor.<name>_opening` entity is created.

## ZHA diagnostics (live device)

<details><summary><code>diagnostics for _TZ3000_bsvqrxru / TS0202 on endpoint 1</code></summary>

```json
{
  "custom_components": {
    "dreame_vacuum": {
      "documentation": "https://github.com/Tasshack/dreame-vacuum/tree/dev?tab=readme-ov-file#features",
      "requirements": [
        "pillow",
        "numpy",
        "pybase64",
        "requests",
        "pycryptodome",
        "python-miio",
        "mini-racer",
        "paho-mqtt"
      ],
      "version": "v2.0.0b20"
    },
    "smartthinq_sensors": {
      "documentation": "https://github.com/ollo69/ha-smartthinq-sensors",
      "requirements": [
        "pycountry>=23.12.11",
        "xmltodict>=0.13.0",
        "charset_normalizer>=3.2.0"
      ],
      "version": "0.41.1"
    }
  },
  "data": {
    "active_coordinator": false,
    "available": true,
    "device_type": "EndDevice",
    "endpoints": {
      "1": {
        "device_type": {
          "id": 1026,
          "name": "IAS_ZONE"
        },
        "in_clusters": [
          {
            "attributes": [
              {
                "id": "0x0001",
                "name": "app_version",
                "value": 70,
                "zcl_type": "uint8"
              },
              {
                "id": "0x0004",
                "name": "manufacturer",
                "value": "_TZ3000_bsvqrxru",
                "zcl_type": "string"
              },
              {
                "id": "0x0005",
                "name": "model",
                "value": "TS0202",
                "zcl_type": "string"
              },
              {
                "id": "0x0007",
                "name": "power_source",
                "value": 3,
                "zcl_type": "enum8"
              },
              {
                "id": "0xfffe",
                "name": "reporting_status",
                "value": 0,
                "zcl_type": "enum8"
              },
              {
                "id": "0x0000",
                "name": "zcl_version",
                "value": 3,
                "zcl_type": "uint8"
              }
            ],
            "cluster_id": "0x0000",
            "endpoint_attribute": "basic"
          },
          {
            "attributes": [
              {
                "id": "0x0021",
                "name": "battery_percentage_remaining",
                "value": 200,
                "zcl_type": "uint8"
              },
              {
                "id": "0x0033",
                "name": "battery_quantity",
                "value": 2,
                "zcl_type": "uint8"
              },
              {
                "id": "0x0034",
                "name": "battery_rated_voltage",
                "value": 15,
                "zcl_type": "uint8"
              },
              {
                "id": "0x0031",
                "name": "battery_size",
                "value": 4,
                "zcl_type": "enum8"
              },
              {
                "id": "0x0020",
                "name": "battery_voltage",
                "value": 30,
                "zcl_type": "uint8"
              }
            ],
            "cluster_id": "0x0001",
            "endpoint_attribute": "power"
          },
          {
            "attributes": [],
            "cluster_id": "0x0003",
            "endpoint_attribute": "identify"
          },
          {
            "attributes": [
              {
                "id": "0x0013",
                "name": "current_zone_sensitivity_level",
                "value": 1,
                "zcl_type": "uint8"
              },
              {
                "id": "0xfff0",
                "name": "motion_timeout",
                "value": 30,
                "zcl_type": "uint16"
              },
              {
                "id": "0x0002",
                "name": "zone_status",
                "value": 1,
                "zcl_type": "map16"
              },
              {
                "id": "0x0001",
                "name": "zone_type",
                "value": 13,
                "zcl_type": "enum16"
              }
            ],
            "cluster_id": "0x0500",
            "endpoint_attribute": "ias_zone"
          }
        ],
        "out_clusters": [
          {
            "attributes": [],
            "cluster_id": "0x000a",
            "endpoint_attribute": "time"
          },
          {
            "attributes": [
              {
                "id": "0x0002",
                "name": "current_file_version",
                "value": 70,
                "zcl_type": "uint32"
              }
            ],
            "cluster_id": "0x0019",
            "endpoint_attribute": "ota",
            "last_query_cmd": {
              "current_file_version": 70,
              "hardware_version": null,
              "image_type": 54179,
              "manufacturer_code": 4417
            }
          },
          {
            "attributes": [],
            "cluster_id": "0x1000",
            "endpoint_attribute": "lightlink"
          }
        ],
        "profile_id": 260
      }
    },
    "exposes_features": [],
    "friendly_manufacturer": "_TZ3000_bsvqrxru",
    "friendly_model": "TS0202",
    "ieee": "**REDACTED**",
    "last_seen": "2026-04-20T16:10:13.722491+00:00",
    "lqi": 160,
    "manufacturer": "_TZ3000_bsvqrxru",
    "manufacturer_code": 4417,
    "model": "TS0202",
    "name": "_TZ3000_bsvqrxru TS0202",
    "neighbors": [],
    "node_descriptor": {
      "aps_flags": 0,
      "complex_descriptor_available": false,
      "descriptor_capability_field": 0,
      "frequency_band": 8,
      "logical_type": "EndDevice",
      "mac_capability_flags": 128,
      "manufacturer_code": 4417,
      "maximum_buffer_size": 66,
      "maximum_incoming_transfer_size": 66,
      "maximum_outgoing_transfer_size": 66,
      "reserved": 0,
      "server_mask": 10752,
      "user_descriptor_available": false
    },
    "nwk": "0xE639",
    "original_signature": {
      "endpoints": {
        "1": {
          "device_type": "0x0402",
          "input_clusters": [
            "0x0001",
            "0x0500",
            "0x0003",
            "0x0000"
          ],
          "output_clusters": [
            "0x1000",
            "0x0006",
            "0x0019",
            "0x000a"
          ],
          "profile_id": "0x0104"
        }
      },
      "manufacturer": "_TZ3000_bsvqrxru",
      "model": "TS0202",
      "node_desc": {
        "aps_flags": 0,
        "complex_descriptor_available": 0,
        "descriptor_capability_field": 0,
        "frequency_band": 8,
        "logical_type": 2,
        "mac_capability_flags": 128,
        "manufacturer_code": 4417,
        "maximum_buffer_size": 66,
        "maximum_incoming_transfer_size": 66,
        "maximum_outgoing_transfer_size": 66,
        "reserved": 0,
        "server_mask": 10752,
        "user_descriptor_available": 0
      }
    },
    "power_source": "Battery or Unknown",
    "quirk_applied": true,
    "quirk_class": "zhaquirks.tuya.builder.EnchantedDeviceV2",
    "routes": [],
    "rssi": null,
    "version": 1,
    "zha_lib_entities": {
      "binary_sensor": [
        {
          "info_object": {
            "attribute_name": "zone_status",
            "available": true,
            "class_name": "IASZone",
            "cluster_handlers": [
              {
                "class_name": "IASZoneClusterHandler",
                "cluster": {
                  "id": 1280,
                  "name": "IAS Zone",
                  "type": "server"
                },
                "endpoint_id": 1,
                "generic_id": "cluster_handler_0x0500",
                "id": "1:0x0500",
                "status": "INITIALIZED",
                "unique_id": "**REDACTED**",
                "value_attribute": null
              }
            ],
            "device_class": "motion",
            "device_ieee": "**REDACTED**",
            "enabled": true,
            "endpoint_id": 1,
            "entity_category": null,
            "entity_registry_enabled_default": true,
            "fallback_name": null,
            "group_id": null,
            "migrate_unique_ids": [],
            "platform": "binary_sensor",
            "primary": true,
            "state_class": null,
            "translation_key": null,
            "translation_placeholders": null,
            "unique_id": "**REDACTED**"
          },
          "state": {
            "available": true,
            "class_name": "IASZone",
            "state": true
          }
        }
      ],
      "button": [
        {
          "info_object": {
            "args": [
              5
            ],
            "available": true,
            "class_name": "IdentifyButton",
            "cluster_handlers": [
              {
                "class_name": "IdentifyClusterHandler",
                "cluster": {
                  "id": 3,
                  "name": "Identify",
                  "type": "server"
                },
                "endpoint_id": 1,
                "generic_id": "cluster_handler_0x0003",
                "id": "1:0x0003",
                "status": "INITIALIZED",
                "unique_id": "**REDACTED**",
                "value_attribute": null
              }
            ],
            "command": "identify",
            "device_class": "identify",
            "device_ieee": "**REDACTED**",
            "enabled": true,
            "endpoint_id": 1,
            "entity_category": "diagnostic",
            "entity_registry_enabled_default": true,
            "fallback_name": null,
            "group_id": null,
            "kwargs": {},
            "migrate_unique_ids": [],
            "platform": "button",
            "primary": false,
            "state_class": null,
            "translation_key": null,
            "translation_placeholders": null,
            "unique_id": "**REDACTED**"
          },
          "state": {
            "available": true,
            "class_name": "IdentifyButton"
          }
        }
      ],
      "number": [
        {
          "info_object": {
            "available": true,
            "class_name": "NumberConfigurationEntity",
            "cluster_handlers": [
              {
                "class_name": "IASZoneClusterHandler",
                "cluster": {
                  "id": 1280,
                  "name": "IAS Zone",
                  "type": "server"
                },
                "endpoint_id": 1,
                "generic_id": "cluster_handler_0x0500",
                "id": "1:0x0500",
                "status": "INITIALIZED",
                "unique_id": "**REDACTED**",
                "value_attribute": null
              }
            ],
            "device_class": null,
            "device_ieee": "**REDACTED**",
            "enabled": true,
            "endpoint_id": 1,
            "entity_category": "config",
            "entity_registry_enabled_default": true,
            "fallback_name": "Motion clear timeout",
            "group_id": null,
            "migrate_unique_ids": [],
            "mode": "auto",
            "native_max_value": 900,
            "native_min_value": 5,
            "native_step": 5,
            "native_unit_of_measurement": "s",
            "platform": "number",
            "primary": false,
            "state_class": null,
            "translation_key": "motion_timeout",
            "translation_placeholders": null,
            "unique_id": "**REDACTED**"
          },
          "state": {
            "available": true,
            "class_name": "NumberConfigurationEntity",
            "state": 30
          }
        }
      ],
      "sensor": [
        {
          "info_object": {
            "available": true,
            "class_name": "LQISensor",
            "cluster_handlers": [
              {
                "class_name": "BasicClusterHandler",
                "cluster": {
                  "id": 0,
                  "name": "Basic",
                  "type": "server"
                },
                "endpoint_id": 1,
                "generic_id": "cluster_handler_0x0000",
                "id": "1:0x0000",
                "status": "INITIALIZED",
                "unique_id": "**REDACTED**",
                "value_attribute": null
              }
            ],
            "device_class": null,
            "device_ieee": "**REDACTED**",
            "enabled": true,
            "endpoint_id": 1,
            "entity_category": "diagnostic",
            "entity_registry_enabled_default": false,
            "fallback_name": null,
            "group_id": null,
            "migrate_unique_ids": [],
            "platform": "sensor",
            "primary": false,
            "state_class": "measurement",
            "suggested_display_precision": null,
            "translation_key": "lqi",
            "translation_placeholders": null,
            "unique_id": "**REDACTED**",
            "unit": null
          },
          "state": {
            "available": true,
            "class_name": "LQISensor",
            "state": 160
          }
        },
        {
          "info_object": {
            "available": true,
            "class_name": "RSSISensor",
            "cluster_handlers": [
              {
                "class_name": "BasicClusterHandler",
                "cluster": {
                  "id": 0,
                  "name": "Basic",
                  "type": "server"
                },
                "endpoint_id": 1,
                "generic_id": "cluster_handler_0x0000",
                "id": "1:0x0000",
                "status": "INITIALIZED",
                "unique_id": "**REDACTED**",
                "value_attribute": null
              }
            ],
            "device_class": "signal_strength",
            "device_ieee": "**REDACTED**",
            "enabled": true,
            "endpoint_id": 1,
            "entity_category": "diagnostic",
            "entity_registry_enabled_default": false,
            "fallback_name": null,
            "group_id": null,
            "migrate_unique_ids": [],
            "platform": "sensor",
            "primary": false,
            "state_class": "measurement",
            "suggested_display_precision": null,
            "translation_key": "rssi",
            "translation_placeholders": null,
            "unique_id": "**REDACTED**",
            "unit": "dBm"
          },
          "state": {
            "available": true,
            "class_name": "RSSISensor",
            "state": null
          }
        },
        {
          "extra_state_attributes": [
            "battery_quantity",
            "battery_size",
            "battery_voltage"
          ],
          "info_object": {
            "available": true,
            "class_name": "Battery",
            "cluster_handlers": [
              {
                "class_name": "PowerConfigurationClusterHandler",
                "cluster": {
                  "id": 1,
                  "name": "Power Configuration",
                  "type": "server"
                },
                "endpoint_id": 1,
                "generic_id": "cluster_handler_0x0001",
                "id": "1:0x0001",
                "status": "INITIALIZED",
                "unique_id": "**REDACTED**",
                "value_attribute": "battery_voltage"
              }
            ],
            "device_class": "battery",
            "device_ieee": "**REDACTED**",
            "enabled": true,
            "endpoint_id": 1,
            "entity_category": "diagnostic",
            "entity_registry_enabled_default": true,
            "fallback_name": null,
            "group_id": null,
            "migrate_unique_ids": [],
            "platform": "sensor",
            "primary": false,
            "state_class": "measurement",
            "suggested_display_precision": 0,
            "translation_key": null,
            "translation_placeholders": null,
            "unique_id": "**REDACTED**",
            "unit": "%"
          },
          "state": {
            "available": true,
            "battery_quantity": 2,
            "battery_size": "AAA",
            "battery_voltage": 3.0,
            "class_name": "Battery",
            "state": 100.0
          }
        }
      ],
      "update": [
        {
          "info_object": {
            "available": true,
            "class_name": "FirmwareUpdateEntity",
            "cluster_handlers": [
              {
                "class_name": "OtaClientClusterHandler",
                "cluster": {
                  "id": 25,
                  "name": "Ota",
                  "type": "client"
                },
                "endpoint_id": 1,
                "generic_id": "cluster_handler_0x0019_client",
                "id": "1:0x0019_client",
                "status": "INITIALIZED",
                "unique_id": "**REDACTED**",
                "value_attribute": null
              }
            ],
            "device_class": "firmware",
            "device_ieee": "**REDACTED**",
            "enabled": true,
            "endpoint_id": 1,
            "entity_category": "config",
            "entity_registry_enabled_default": true,
            "fallback_name": null,
            "group_id": null,
            "migrate_unique_ids": [],
            "platform": "update",
            "primary": false,
            "state_class": null,
            "supported_features": 7,
            "translation_key": null,
            "translation_placeholders": null,
            "unique_id": "**REDACTED**"
          },
          "state": {
            "available": true,
            "class_name": "FirmwareUpdateEntity",
            "in_progress": false,
            "installed_version": "0x00000046",
            "latest_version": null,
            "release_notes": null,
            "release_summary": null,
            "release_url": null,
            "update_percentage": null
          }
        }
      ]
    }
  },
  "home_assistant": {
    "arch": "x86_64",
    "container_arch": "amd64",
    "dev": false,
    "docker": true,
    "hassio": false,
    "installation_type": "Home Assistant Container",
    "os_name": "Linux",
    "os_version": "6.14.0-37-generic",
    "python_version": "3.14.2",
    "run_as_root": true,
    "timezone": "Europe/Brussels",
    "version": "2026.4.1",
    "virtualenv": false
  },
  "integration_manifest": {
    "after_dependencies": [
      "hassio",
      "onboarding",
      "usb"
    ],
    "codeowners": [
      "dmulcahey",
      "adminiuga",
      "puddly",
      "TheJulianJES"
    ],
    "config_flow": true,
    "dependencies": [
      "file_upload",
      "homeassistant_hardware"
    ],
    "documentation": "https://www.home-assistant.io/integrations/zha",
    "domain": "zha",
    "integration_type": "hub",
    "iot_class": "local_polling",
    "is_built_in": true,
    "loggers": [
      "aiosqlite",
      "bellows",
      "crccheck",
      "pure_pcapy3",
      "zhaquirks",
      "zigpy",
      "zigpy_deconz",
      "zigpy_xbee",
      "zigpy_zigate",
      "zigpy_znp",
      "zha",
      "universal_silabs_flasher",
      "serialx"
    ],
    "name": "Zigbee Home Automation",
    "overwrites_built_in": false,
    "requirements": [
      "zha==1.1.1",
      "serialx==0.6.2"
    ],
    "usb": [
      {
        "description": "*2652*",
        "known_devices": [
          "slae.sh cc2652rb stick"
        ],
        "pid": "EA60",
        "vid": "10C4"
      },
      {
        "description": "*slzb-07*",
        "known_devices": [
          "smlight slzb-07"
        ],
        "pid": "EA60",
        "vid": "10C4"
      },
      {
        "description": "*sonoff*plus*",
        "known_devices": [
          "sonoff zigbee dongle plus v2"
        ],
        "pid": "55D4",
        "vid": "1A86"
      },
      {
        "description": "*sonoff*plus*",
        "known_devices": [
          "sonoff zigbee dongle plus"
        ],
        "pid": "EA60",
        "vid": "10C4"
      },
      {
        "description": "*tubeszb*",
        "known_devices": [
          "TubesZB Coordinator"
        ],
        "pid": "EA60",
        "vid": "10C4"
      },
      {
        "description": "*tubeszb*",
        "known_devices": [
          "TubesZB Coordinator"
        ],
        "pid": "7523",
        "vid": "1A86"
      },
      {
        "description": "*zigstar*",
        "known_devices": [
          "ZigStar Coordinators"
        ],
        "pid": "7523",
        "vid": "1A86"
      },
      {
        "description": "*conbee*",
        "known_devices": [
          "Conbee II"
        ],
        "pid": "0030",
        "vid": "1CF1"
      },
      {
        "description": "*conbee*",
        "known_devices": [
          "Conbee III"
        ],
        "pid": "6015",
        "vid": "0403"
      },
      {
        "description": "*zigbee*",
        "known_devices": [
          "Nortek HUSBZB-1"
        ],
        "pid": "8A2A",
        "vid": "10C4"
      },
      {
        "description": "*zigate*",
        "known_devices": [
          "ZiGate+"
        ],
        "pid": "6015",
        "vid": "0403"
      },
      {
        "description": "*zigate*",
        "known_devices": [
          "ZiGate"
        ],
        "pid": "EA60",
        "vid": "10C4"
      },
      {
        "description": "*bv 2010/10*",
        "known_devices": [
          "Bitron Video AV2010/10"
        ],
        "pid": "8B34",
        "vid": "10C4"
      },
      {
        "description": "*sonoff*max*",
        "known_devices": [
          "SONOFF Dongle Max MG24"
        ],
        "pid": "EA60",
        "vid": "10C4"
      },
      {
        "description": "*sonoff*lite*mg21*",
        "known_devices": [
          "sonoff zigbee dongle lite mg21"
        ],
        "pid": "EA60",
        "vid": "10C4"
      }
    ],
    "zeroconf": [
      {
        "name": "tube*",
        "type": "_esphomelib._tcp.local."
      },
      {
        "name": "*zigate*",
        "type": "_zigate-zigbee-gateway._tcp.local."
      },
      {
        "name": "*zigstar*",
        "type": "_zigstar_gw._tcp.local."
      },
      {
        "name": "uzg-01*",
        "type": "_uzg-01._tcp.local."
      },
      {
        "name": "slzb-06*",
        "type": "_slzb-06._tcp.local."
      },
      {
        "name": "xzg*",
        "type": "_xzg._tcp.local."
      },
      {
        "name": "czc*",
        "type": "_czc._tcp.local."
      },
      {
        "name": "*",
        "type": "_zigbee-coordinator._tcp.local."
      }
    ]
  },
  "issues": [],
  "setup_times": {
    "01JQ4ER63KH4KZBMR5D7B9VW0G": {
      "config_entry_setup": 13.1637274089735,
      "wait_import_platforms": -0.016411149874329567
    },
    "null": {
      "setup": 4.594912752509117e-05
    }
  }
}
```

</details>

### CODE REVIEW

File: zhaquirks/tuya/ts0202.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            if self._timer_handle:
+                self._timer_handle.cancel()
+            self._timer_handle = self._loop.call_later(self.reset_s, self._turn_off)
+        super()._update_attribute(attrid, value)
+
+    async def write_attributes(
+        self,
+        attributes: dict[str | int, Any],
+        manufacturer: int | None = None,
+    ) -> list:
```
[the remark this answers is not in the document]
Comment (vjonas) [reply zhaquirks/tuya/ts0202.py]: Fixed in fabff84. The signature now matches the canonical `LocalDataCluster.write_attributes` pattern: `attributes: dict[str | int | foundation.ZCLAttributeDef, Any]`, `manufacturer: int | UndefinedType | None = UNDEFINED`, `**kwargs`, returning `list[list[foundation.WriteAttributesStatusRecord]]`. `manufacturer` and any extra kwargs are forwarded to `super().write_attributes` unchanged. Two new tests cover the `ZCLAttributeDef`-keyed write path and verify that `manufacturer` + arbitrary kwargs (e.g. `update_cache=False`) make it through to the base cluster.

File: zhaquirks/tuya/ts0202.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    * Pins ``zone_type = Motion_Sensor`` so the binary_sensor renders as
+      ``device_class: motion`` without waiting for the device to answer.
+    * Schedules the auto-clear timer for *both* ``zone_status_change_notification``
+      commands (inherited from ``MotionWithReset``) and ``zone_status``
+      attribute reports, since these TS0202 clones mix the two paths across
+      firmware revisions.
+    * Exposes a synthetic, writable ``motion_timeout`` attribute (seconds)
+      that backs a Number entity. Writing it updates the auto-clear timer
+      live; no Zigbee frames are sent to the device since this firmware has
+      no runtime-configurable keep_time.
```
[the remark this answers is not in the document]
Comment (vjonas) [reply zhaquirks/tuya/ts0202.py]: Fixed in fabff84. Picked the "make it actually live" option: when `motion_timeout` is written and `_timer_handle is not None` (motion is currently active), the running auto-clear timer is cancelled and a fresh one is scheduled with the new value measured from now. If no timer is pending the value is just stashed in `reset_s` and used on the next motion event. Two new tests lock this in: `test_writing_motion_timeout_during_active_motion_reschedules_timer_live` (active path) and `test_writing_motion_timeout_with_no_active_motion_does_not_arm_timer` (no-active-motion guard). Docstring rewritten to describe both behaviours.

**Comment by vjonas** [comment]
should be updated now.
